token/metadata.go

```diff
@@ -43,124 +48,135 @@ func (m *Metadata) SpentTokenID() []*token.ID {
 // - The senders are included if and only if there is at least one output whose owner has the given enrollment IDs.
 // Application metadata is always included
 func (m *Metadata) FilterBy(eIDs ...string) (*Metadata, error) {
-	res := &Metadata{
+	if len(eIDs) == 0 {
+		return m, nil
+	}
+
+	clonedMetadata := &Metadata{
 		TokenService:         m.TokenService,
 		WalletService:        m.WalletService,
 		TokenRequestMetadata: &driver.TokenRequestMetadata{},
 		Logger:               m.Logger,
 	}
-
+	eIDSet := collections.NewSet(eIDs...)
```

Comment:
We can split this method to smaller pieces
```
issues, err := m.filterIssues(m.TokenRequestMetadata.Issues, eIDSet)
transfers, err := m.filterTransfers(m.TokenRequestMetadata.Transfers, eIDSet)
return &Metadata{
 TokenService: m.TokenService,
 WalletService: m.WalletService,
 Logger: m.Logger,
 TokenRequestMetadata: &driver.TokenRequestMetadata{
  Issues: issues,
  Transfers: transfers,
  Application: m.TokenRequestMetadata.Application
 }
}, nil
```

Reply:
okay